### PR TITLE
Add autocompletion for composer recipes commands

### PR DESCRIPTION
**Pull request by duboiss**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
No autocompletion.

**What is the new behavior (if this is a feature change)?**
Two new commands which only appear in the presence of a `symfony.lock` file.

**Additional info:**
This is a composer plugin specific to the symfony framework (symfony flex).

### CODE REVIEW

File: dev/composer.ts [hunk 1]
```diff
@@ -149,6 +154,84 @@ const completionSpec: Fig.Spec = {
...
+
+      const recipesCommonOptions = [
+        { name: ["-h", "--help"], description: "Display this help message" },
+        { name: ["-q", "--quiet"], description: "Do not output any message" },
+        {
+          name: ["-V", "--version"],
+          description: "Display this application version",
+        },
+        { name: "--ansi", description: "Force ANSI output" },
+        { name: "--no-ansi", description: "Disable ANSI output" },
```
Comment (Monkeyank) [inline dev/composer.ts:166]: Can we add an `exclusiveOn: ` prop

File: dev/composer.ts [hunk 2]
```diff
@@ -149,6 +154,84 @@ const completionSpec: Fig.Spec = {
...
+        {
+          name: ["-n", "--no-interaction"],
+          description: "Do not ask any interactive question",
+        },
+        {
+          name: "--profile",
+          description: "Display timing and memory usage information",
+        },
+        { name: "--no-plugins", description: "Whether to disable plugins" },
```
Comment (Monkeyank) [inline dev/composer.ts:175]: Same as above

**Review by Monkeyank — CHANGES_REQUESTED** [review]

File: dev/composer.ts [hunk 3]
```diff
@@ -149,6 +154,84 @@ const completionSpec: Fig.Spec = {
           }),
         });
       }
+
+      const recipesCommonOptions = [
```
Comment (QuiiBz) [inline dev/composer.ts:158]: Could you add a type of `Fig.Option[]` here?

File: dev/composer.ts [hunk 4]
```diff
@@ -149,6 +154,84 @@ const completionSpec: Fig.Spec = {
...
+        },
+        {
+          name: "--profile",
+          description: "Display timing and memory usage information",
+        },
+        { name: "--no-plugins", description: "Whether to disable plugins" },
+        {
+          name: ["-d", "--working-dir"],
+          description:
+            "If specified, use the given directory as working directory",
```
Comment (QuiiBz) [inline dev/composer.ts:179]: Does this option take an argument, which is the working directory?

File: dev/composer.ts [hunk 5]
```diff
@@ -149,6 +154,84 @@ const completionSpec: Fig.Spec = {
...
+          args: {
+            name: "packages",
+            description: "Recipes that should be installed",
+            isVariadic: true,
+          },
+          options: [
+            {
+              name: "--force",
+              description:
+                "Overwrite existing files when a new version of a recipe is available",
```
Comment (QuiiBz) [inline dev/composer.ts:229]: We can definitely add [isDangerous](https://fig.io/docs/reference/option#isdangerous) here!

**Review by QuiiBz — CHANGES_REQUESTED** [review]

File: dev/composer.ts (hunk 4 above)
Comment (duboiss) [reply dev/composer.ts:179]: We can use :
```
-d <path>
--working-dir=<path>
--working-dir <path>
```

File: dev/composer.ts (hunk 2 above)
Comment (duboiss) [reply dev/composer.ts:183]: For which command?

File: dev/composer.ts [hunk 6]
```diff
@@ -149,6 +154,93 @@ const completionSpec: Fig.Spec = {
...
+        {
+          name: "--profile",
+          description: "Display timing and memory usage information",
+        },
+        { name: "--no-plugins", description: "Whether to disable plugins" },
+        {
+          name: ["-d", "--working-dir"],
+          description:
+            "If specified, use the given directory as working directory",
+        },
```
Comment (SeparateRecords) [inline dev/composer.ts:188]:
```suggestion
        {
          name: ["-d", "--working-dir"],
          description:
            "If specified, use the given directory as working directory",
          args: {
            name: "dir",
            template: "folders",
          },
        },
```

File: dev/composer.ts [hunk 7]
```diff
@@ -149,6 +154,93 @@ const completionSpec: Fig.Spec = {
...
+        { name: "--no-cache", description: "Prevent use of the cache" },
+        {
+          name: ["-v", "-vv", "-vvv", "--verbose"],
+          description:
+            "Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug",
+        },
+      ];
+
+      if (!symfonyLock.endsWith("(No such file or directory)")) {
```
Comment (SeparateRecords) [inline dev/composer.ts:197]: Nitpicky suggestion: extracting this to a separate variable would make the intention a lot clearer
<sub><i>and yes, I do see the irony, given that I'm the one who suggested this code on discord 😅</i></sub>

```suggestion
      const lockExists = !symfonyLock.endsWith("(No such file or directory)");
      if (lockExists) {
```

File: dev/composer.ts [hunk 8]
```diff
@@ -149,6 +154,100 @@ const completionSpec: Fig.Spec = {
...
+          description:
+            "If specified, use the given directory as working directory",
+          args: {
+            name: "dir",
+            template: "folders",
+          },
+        },
+        { name: "--no-cache", description: "Prevent use of the cache" },
+        {
+          name: ["-v", "-vv", "-vvv", "--verbose"],
```
Comment (fedeci) [inline dev/composer.ts:195]: These are different options about verbosity. We can'y join all of them together.

**Review by fedeci — CHANGES_REQUESTED** [review]

File: dev/composer.ts [hunk 9]
```diff
@@ -149,6 +154,100 @@ const completionSpec: Fig.Spec = {
...
+          description:
+            "Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug",
+        },
```
Comment (QuiiBz) [inline dev/composer.ts:198]: As @fedeci mentioned, this should be separated in 3 options: normal (v), verbose (vv) and debug (vvv)

**Review by QuiiBz — CHANGES_REQUESTED** [review]

**Review by fedeci — APPROVED** [review]

**Review by QuiiBz — APPROVED** [review]
Thank you so much for this PR, LGTM 🚀

**Pull request merged**